src/libraries/System.Runtime.CompilerServices.Unsafe/tests/UnsafeTests.cs

```diff
@@ -53,7 +53,15 @@ public static unsafe void WriteBytesIntoInt32()
             Assert.Equal(2, b4.B2);
             Assert.Equal(3, b4.B3);
 
-            int expected = (b4.B3 << 24) + (b4.B2 << 16) + (b4.B1 << 8) + (b4.B0);
+            int expected;
+            if (BitConverter.IsLittleEndian)
+            {
+                expected = (b4.B3 << 24) + (b4.B2 << 16) + (b4.B1 << 8) + (b4.B0);
+            }
+            else
+            {
+                expected = (b4.B0 << 24) + (b4.B1 << 16) + (b4.B2 << 8) + (b4.B3);
```

Comment:
I wonder if we should instead have a test helper method that turns int32 and int64 from littleEndian to bigendian so that we don't have to duplicate shifting or checks and the expected value is only really ever written once in our tests. Similar to what you are doing in the BitConverteArray.cs tests where you just save to an array and then reverse it.

Reply:
Well, in BitConverterArray.cs it is more obvious how do those swaps, given that we're really operating on single scalar values (or arrays of the same type).

Here, the tests deliberately overlay multiple data types (modifying single bytes in a long, reading a pair of bytes as a short etc.), which seem inherently endian-sensitive.  Since the intent of this test is exactly to verify that those aliased manipulations work as expected (where the "expected" behavior is explicitly endian-sensitive), it seemed to make more sense to me to verify the little- and big-endian results separately here.

If you have any specific suggestion of how to structure the tests differently, I'd be happy to implement and test ...